.github/workflows/tests.yml

```diff
@@ -131,33 +131,31 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        # 08/06/2022 - Disable Thread sanitiser build as it exhausts all the
-        # memory available in GHA runners
-        sanitiser: [None, Address, Undefined, Leak]
+        # 21/06/2022 - We have disabled the ASAN and TSAN builds as they were
+        # consuming upwards of 15 to 20 GB of memory (each). Even on a self-
+        # hosted runner this is too much
+        sanitiser: [None, Address, Thread, Undefined, Leak]
```

Comment:
Before merging this, can you remove all sanitiser-related changes into the other sanitiser PR?

Reply:
Yes, I'm kinda stuck in the sanitisers PR, so I will work on that first. Then rebase and merge this in.